app/src/main/java/com/android_academy/covid_19/ui/map/MapManager.kt

```diff
@@ -84,8 +86,6 @@ class MapManagerImpl(
             return
         }
 
-        map.clear()
```

Comment:
this line was clearing the map from "previous" selected makers. For example if I am looking using the filter for a specific time frame I see 20 markers, moving to a different timeframe I should see the new markers and not the previous ones.

Reply:
It was wrong. It cleared all markers and not only those that you wanted. Actually it cause all markers to disappear regardless of that type that changes were infected.
Here you need to clear markers only of type Infected